ligand/views.py

```diff
@@ -44,26 +112,757 @@ class LigandNameSelection(AbsTargetSelection):
+        with connection.cursor() as cursor:
+            try:
+                if smiles_only:
+                    cursor.execute('SELECT mol_from_smiles(%s);', [smarts])
+                    
+                else:
+                    cursor.execute('SELECT qmol(%s);', [smarts])
+                    
+            except DataException:
+                msg = invalid_smarts_msg
+                error = True
+                return (error, msg)
+            r = cursor.fetchall()
+            
+            if len(r) < 1:
+                msg = invalid_smarts_msg
+                error = True
+                return (error, msg)
+            if r[0] is None:
+                msg = invalid_smarts_msg
+                error = True
+                return (error, msg)
+                
+        return (error, msg)
+    
+class ReadInputLigandBulkSearch(View):
+    min_similarity = None
+    OK_request_dict = {'status':'OK',
+                    'status_code':200,
+                    'reason_phrase':'OK',
+                    'msg':'OK'}
+    bad_request_dict = {'status':'error',
+                    'status_code':400,
+                    'reason_phrase':'Bad Request',
+                    'msg':''}
+    search_params_data_keys = ['search_type','search_text','stereochemistry','field']
+
+    def validate_SMARTS(self,smarts,smiles_only=False):
+        return validate_SMARTS(smarts,smiles_only=smiles_only)
+  
+    def get(self, request, *args, **kwargs):
+        OK_request_dict = self.OK_request_dict.copy()
+        bad_request_dict = self.bad_request_dict.copy()
+        OK_request_dict['csrf_token'] = get_token(request)
+        return JsonResponse(OK_request_dict)
+
+    def post(self, request, *args, **kwargs):
+        OK_request_dict = self.OK_request_dict.copy()
+        bad_request_dict = self.bad_request_dict.copy()
+        search_params_data = {}
+        ligand_chemical_structure_search = False
+        error = False
+
+        search_type = request.POST.get('search_type',None)
+        if search_type in {'smiles','smarts','inchikey'}:
+            ligand_chemical_structure_search = True
+
+            selection_ligand_chemical_structure_search_search_type_selection = value = request.POST.get('selection_ligand_chemical_structure_search_search_type_selection',None)
+            error = not validate_selection_ligand_chemical_structure_search_search_type_selection(value)
+            msg = 'Invalid selection_ligand_chemical_structure_search_search_type_selection value: '+str(value)
+            if error:
+                request.session['ligand_bulk_search_error_msg'] = msg
+                request.session.modified = True
+                bad_request_dict['msg'] = msg
+                return JsonResponse(bad_request_dict,status=400,reason='Bad Request')
+            del value
+
+        for key in self.search_params_data_keys:
+            if key == 'search_text' and search_type != 'name':
+                # if search_type == 'id':
```

Comment:
please remove commented lines if not needed

Reply:
Solved in [d47cf16](https://github.com/protwis/protwis/pull/550/commits/d47cf16425ca86f6174af64c3ef73aa50cee391e).